core/src/test/java/org/elasticsearch/VersionTests.java

```diff
@@ -343,8 +343,6 @@ public void testIsCompatible() {
         } else {
             assertFalse(isCompatible(Version.CURRENT, Version.fromString("7.0.0")));
         }
-        assertFalse("only compatible with the latest minor",
```

Comment:
why do we remove this assertion, I think it's valid?

Reply:
Sorry, yes, you're right it's valid. There's a bug somewhere. Hunting it now...